lib/KBaseReport/utils/file_utils.py

```diff
@@ -11,17 +11,20 @@
 """
 
 
-def fetch_or_upload_file_links(dfu, files):
+def fetch_or_upload_file_links(dfu, files, templater):
     """
     Given a list of dictionaries of files for the `file_links` parameter in an extended_report
     Fetch by shock ID or upload the file or zipped directory
     :param dfu: DataFileUtil client instance
+    :param templater: TemplateUtil instance
     :param files: list of file dictionaries (having the File type from the KIDL spec)
     :return: list of file dictionaries that that can be uploaded to the workspace for the report
     """
     out_files = []
-    validate_files(files)  # Assures that every file has either a 'path' or 'shock_id'
     for each_file in files:
+        if 'template' in each_file:
+            each_file = _render_template_add_path(templater, each_file)
+
```

Comment:
Renders the template to a temporary file and puts the temp file path in the `path` dict entry.